nyx_client/nyx_client/client.py

```diff
@@ -797,3 +866,95 @@ def unsubscribe(self, data: Data):
         # Creator is expected to be double encoded
         creator = quote_plus(quote_plus(data.creator))
         self._nyx_delete(f"{NYX_PURCHASES_TRANSACTIONS_ENDPOINT}/{creator}/{data.name}")
+
+    def organizations(self) -> list[RemoteHost]:
+        """Get a list of all organizations in the federated network.
+
+        Returns:
+            A list of `RemoteHost` objects
+
+        Raises:
+            requests.HTTPError: if the API request fails.
+        """
+        raw_orgs = self._nyx_get(NYX_ORG_ENDPOINT)
+        return [RemoteHost.from_dict(org) for org in raw_orgs]
+
+    def get_circles(self) -> list[Circle]:
+        """Get a list of circles.
+
+        Returns:
+            A list of `Circle` objects
+
+        Raises:
+            requests.HTTPError: if the API request fails.
+        """
+        circles_raw = self._nyx_get(NYX_CIRCLE_ENDPOINT)
+        return [Circle.from_dict(r) for r in circles_raw]
+
+    def get_circle_by_name(self, circle_name: str) -> Circle:
+        """Get a list of circles.
+
+        Args:
+            circle_name: The name of the circle to get.
+
+        Returns:
+            A `Circle` object
+
+        Raises:
+            requests.HTTPError: if the API request fails.
+        """
+        circle_raw = self._nyx_get(NYX_CIRCLE_ENDPOINT + "/" + circle_name)
+        return Circle.from_dict(circle_raw)
+
+    def create_circle(self, circle: Circle) -> Circle:
+        """Create a circle.
+
+        Args:
+            circle: The circle to be created.
+
+        Returns:
+            An updated `Circle` object
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        circle_json = circle.as_dict()
+        circle_json.pop("did")
+        resp = self._nyx_post(NYX_CIRCLE_ENDPOINT, data=circle_json)
+
+        circle.did = resp["did"]
+        return circle
+
+    def update_circle(self, circle: Circle):
+        """Updates a circle, based on the cirle's name.
+
+        Args:
+            circle: The circle to be updated.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        circle_json = circle.as_dict()
+        self._nyx_put(f"{NYX_CIRCLE_ENDPOINT}/{circle.name}", data=circle_json)
```

Comment:
(unimportant) - this could present/return the current state like `create` - but maybe not worth it right now